internal/pkg/service/common/etcdop/watch.go

```diff
@@ -371,21 +372,22 @@ func wrapStreamWithRestart(ctx context.Context, channelFactory func(ctx context.
 
 				// Handle initialization error
 				if err := resp.InitErr; err != nil {
-					if init {
+					switch {
+					case init:
 						// Stop on initialization error
 						stream.channel <- resp
 						return
-					} else if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+					case errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded):
 						// Context cancelled event is forwarded only during the initialization.
 						// In other cases, closing the channel is sufficient.
 						continue
-					} else {
+					default:
 						// Convert initialization error
 						// from an 1+ attempt to a common error and restart watch.
 						resp.Err = err
 						resp.InitErr = nil
 						stream.channel <- resp
-						break
+						break ChannelLoop
```

Comment:
> This is the only way to satisfy gocritic here but maybe it would be better to revert this change and just ignore the error?

yes, in this case is IF simpler solution